### PR TITLE
Arb tx receipts

### PR DESCRIPTION
**Pull request by hkalodner**
This PR adds various arbitrum specific data to the RPC interface

- Added to txes
  - `l1SequenceNumber`: This is the sequence number of the batch that included this transaction
  - `parentRequestId`: If this transaction was included in a batch, return the id of that batch or null otherwise
  - `indexInParent`: If the transaction was included in a batch, return the index of this transaction in the batch or null 

- Added to tx receipts
  - `returnCode`: This returns the arbitrum specific return code explaining why the transaction failed
otherwise

### CODE REVIEW

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6417446&bid=20792391)

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6417446&bid=20792391)

File: packages/arb-tx-aggregator/web3/eth.go [hunk 1]
```diff
@@ -410,6 +420,9 @@ func (s *Server) makeTransactionResult(res *evm.TxResult) (*TransactionResult, e
 		V:                (*hexutil.Big)(vVal),
 		R:                math.U256Bytes(rVal),
 		S:                math.U256Bytes(sVal),
+		L1SeqNum:         (*hexutil.Big)(provenance.L1SeqNum),
+		ParentRequestId:  parentRequestId,
+		IndexInParent:    (*hexutil.Big)(provenance.IndexInParent),
```
Comment (fredlacs) [inline packages/arb-tx-aggregator/web3/eth.go:425]: Is the IndexInParent the same value as returned in the TransactionIndex [in the transaction receipt](https://github.com/OffchainLabs/arbitrum/blob/b6856893221c388f84b115bf118bff2078027483/packages/arb-tx-aggregator/web3/eth.go#L254)?

Comment (hkalodner) [reply packages/arb-tx-aggregator/web3/eth.go:425]: Nah, this is the index in the arbitrum batch

**Review by fredlacs — APPROVED** [review]

**Pull request merged**